### PR TITLE
+str #17461 add a parallel version of foreach

### PR DESCRIPTION
**Pull request by agolubev**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala [hunk 1]
```diff
@@ -66,6 +66,9 @@ object Sink {
   def foreach[T](f: function.Procedure[T]): Sink[T, Future[Unit]] =
     new Sink(scaladsl.Sink.foreach(f.apply))
 
+  def foreachParallel[T](parallel: Int)(f: function.Procedure[T])(ec: ExecutionContext): Sink[T, Future[Unit]] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala]: needs javadoc

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala [hunk 2]
```diff
@@ -137,6 +136,48 @@ object Sink extends SinkApply {
...
+   *
+   * If function `f` throws an exception and the supervision decision is
+   * [[akka.stream.Supervision.Resume]] or [[akka.stream.Supervision.Restart]] the
+   * element is dropped and the stream continues.
+   *
+   * @see [[#mapAsyncUnordered]]
+   */
+  def foreachParallel[T](parallelism: Int)(f: T ⇒ Unit)(implicit ec: ExecutionContext): Sink[T, Future[Unit]] = {
+
+    def newForeachParallelStage(): (PushStage[T, Unit], Future[Unit]) = {
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala]: You don't need this anymore, see: https://github.com/akka/akka/pull/17666

**Comment by agolubev** [comment]
I think this is already fixed here: #17658

**Comment by agolubev** [comment]
Did not see anything in the build log related to my commit. @ktoso can you take a look when you have time?

**Comment by drewhk** [comment]
LGTM
The failure is unrelated and seems like just a timing issue (500ms is a bit short for these virtual boxes)

**Comment by 2m** [comment]
Fixes https://github.com/akka/akka/issues/17461

**Pull request merged**